src/Npgsql/Internal/TypeHandling/INpgsqlTypeHandler.cs

```diff
@@ -53,4 +54,22 @@ public interface INpgsqlTypeHandler<T>
     /// An optional token to cancel the asynchronous operation. The default value is <see cref="CancellationToken.None"/>.
     /// </param>
     Task Write([DisallowNull] T value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async, CancellationToken cancellationToken = default);
-}
\ No newline at end of file
+}
+
+static class INpgsqlTypeHandlerExtensions
+{
+    public static async Task WriteWithLength<T>(this INpgsqlTypeHandler<T> handler, T? value, NpgsqlWriteBuffer buf, NpgsqlLengthCache? lengthCache, NpgsqlParameter? parameter, bool async, CancellationToken cancellationToken = default)
```

Comment:
We can't use DIMs on ns2.0

Reply:
Yeah regardless of DIM it might not be what we want to add as a primitive to the interface. It feels more like a helper to me, which is also how it exists on NpgsqlTypeHandler. (doesn't mean much but it also saves a virtual call)